ec/ecresource/deploymentresource/elasticsearchstate/elasticsearch_expanders.go

```diff
@@ -183,3 +198,36 @@ func parseNodeType(topology map[string]interface{}) models.ElasticsearchNodeType
 
 	return result
 }
+
+func expandConfig(raw interface{}) *models.ElasticsearchConfiguration {
+	var res = &models.ElasticsearchConfiguration{}
+	for _, rawCfg := range raw.([]interface{}) {
+		var cfg = rawCfg.(map[string]interface{})
+		if settings, ok := cfg["user_settings_json"]; ok && settings != nil {
+			if s, ok := settings.(string); ok && s != "" {
+				res.UserSettingsJSON = settings
+			}
+		}
+		if settings, ok := cfg["user_settings_override_json"]; ok && settings != nil {
+			if s, ok := settings.(string); ok && s != "" {
+				res.UserSettingsOverrideJSON = settings
+			}
+		}
+		if settings, ok := cfg["user_settings_yaml"]; ok {
+			res.UserSettingsYaml = settings.(string)
+		}
+		if settings, ok := cfg["user_settings_override_yaml"]; ok {
+			res.UserSettingsOverrideYaml = settings.(string)
+		}
+
+		if v, ok := cfg["plugins"]; ok {
+			res.EnabledBuiltInPlugins = util.ItemsToString(v.(*schema.Set).List())
+		}
+	}
+
+	if !reflect.DeepEqual(res, &models.ElasticsearchConfiguration{}) {
```

Comment:
Why is this conditional necessary?

Reply:
So that we don't return a `&models.ElasticsearchConfiguration{}` instead of `nil`. This is to prevent the json serializer to return set these fields to `{}`, which is not the intention. If none of the fields can be expanded, then it means that the object itself is empty, not `{}`. It's a necessary evil I guess, annoying to have to write it like this.

Reply:
ah! gotcha 👌